xml_converter/generators/code_generator.py

```diff
@@ -289,12 +303,14 @@ class AttributeVariable:
     xml_fields: List[str]
     protobuf_field: str
     args: List[str] = field(default_factory=list)
-    default_xml_fields: List[str] = field(default_factory=list)
-    xml_export: str = ""
+    default_xml_field: str = ""
     side_effects: List[str] = field(default_factory=list)
-    compound_name: Optional[str] = None
+    xml_parent_export: List[str] = field(default_factory=list)
+    parser_flag_name: Optional[str] = ""
+    write_to_xml: bool = True
     is_trigger: bool = False
     uses_file_path: bool = False
+    is_component: bool = False
```

Comment:
what if we change these two from
`write_to_xml_field` and `is_component` to something like

`write_as_top_level_xml_attribute` and `write_as_top_level_proto_field`

Reply:
As discussed, this has some side effects and would be its own PR